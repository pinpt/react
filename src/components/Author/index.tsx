export interface IAuthorProps {
	className?: string;
	avatarUrl: string;
	name?: string;
}

const Author = (props: IAuthorProps) => {
	const { className = '', avatarUrl, name } = props;

	return (
		<div className={`Pinpoint Author ${className}`}>
			<img className="avatar" src={avatarUrl} />
			<span className="name">{name}</span>
		</div>
	);
};

export default Author;
