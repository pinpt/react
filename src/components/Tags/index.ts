import Bar from './Bar';
import Item from './Item';

export default {
	Bar,
	Item,
};
