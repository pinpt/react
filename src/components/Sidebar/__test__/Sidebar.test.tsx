import renderer from 'react-test-renderer';
import Sidebar from '..';
import Author from '../../Author';
import Tags from '../../Tags';
import Clap from '../../Clap';

const AVATAR =
	'https://file.pinpoint.com/6cc744395c563f0072b339070aa327f3;UCGt%3D%3DkE~A%3FZJ6%5EhIpS%23o%5BELE3oLT0I%40t7E2;400x400.png?rw=48&rh=48';

test('Test default', () => {
	const component = renderer.create(
		<Sidebar
			author={<Author avatarUrl={AVATAR} name="Keegan" />}
			tags={<Tags.Bar tags={['feature', 'improvement', 'mobile']} />}
			clap={<Clap clapCount={22} handleClap={() => console.log('click!')} />}
		/>
	);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Test no author', () => {
	const component = renderer.create(
		<Sidebar
			tags={<Tags.Bar tags={['feature', 'improvement', 'mobile']} />}
			clap={<Clap clapCount={22} handleClap={() => console.log('click!')} />}
		/>
	);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('clap only', () => {
	const component = renderer.create(
		<Sidebar
			tags={<Tags.Bar tags={['feature', 'improvement', 'mobile']} />}
			clap={<Clap clapCount={22} handleClap={() => console.log('click!')} />}
		/>
	);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Test custom className', () => {
	const component = renderer.create(
		<Sidebar
			author={<Author avatarUrl={AVATAR} name="Keegan" />}
			tags={<Tags.Bar tags={['feature', 'improvement', 'mobile']} />}
			clap={<Clap clapCount={22} handleClap={() => console.log('click!')} />}
			className="custom-test"
		/>
	);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
