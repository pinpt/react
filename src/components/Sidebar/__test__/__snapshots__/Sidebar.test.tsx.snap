// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test custom className 1`] = `
<div
  className="Pinpoint Sidebar custom-test"
>
  <div
    className="Pinpoint Author "
  >
    <span
      aria-label="Keegan avatar"
      className="Pinpoint Internal Link  "
    >
      <img
        alt=""
        className="Pinpoint Image avatar blurhash"
        decoding="async"
        height={48}
        loading="eager"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw=="
        width={48}
      />
      <span
        className="name"
      >
        Keegan
      </span>
    </span>
  </div>
  <div
    className="Pinpoint Tag Bar "
  >
    <div
      className="Pinpoint Tag Item   "
      style={
        Object {
          "backgroundColor": "var(--tag-feature-bgcolor,#B6F5B6)",
          "border": "1px solid var(--tag-bcolor, #3f7a4333)",
          "color": "var(--tag-fgcolor, #27632e)",
        }
      }
    >
      feature
    </div>
    <div
      className="Pinpoint Tag Item   "
      style={
        Object {
          "backgroundColor": "var(--tag-improvement-bgcolor,#a2eeef)",
          "border": "1px solid var(--tag-bcolor, #23747633)",
          "color": "var(--tag-fgcolor, #005e5f)",
        }
      }
    >
      improvement
    </div>
    <div
      className="Pinpoint Tag Item   "
      style={
        Object {
          "backgroundColor": "#f473da",
          "border": "1px solid var(--tag-bcolor, #72006233)",
          "color": "var(--tag-fgcolor, #ffffff)",
        }
      }
    >
      mobile
    </div>
  </div>
  <div
    className="Pinpoint Clap wrapper "
    onClick={[Function]}
  >
    <div
      className="Pinpoint Clap icon "
    >
      <svg
        fill="none"
        height="45"
        viewBox="0 0 985 997"
        width="45"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M384.602 300.771L374.481 280.34C355.482 241.988 303.813 234.105 275.654 266.246C233.852 226.468 167.893 261.376 175.201 316.902C133.778 319.79 107.584 364.684 126.716 403.304L191.394 533.862C167.122 530.597 142.57 542.537 130.768 565.569C115.633 594.94 127.083 631.118 156.274 646.565L345.7 747.352C352.529 750.999 360.713 751.074 367.613 747.656L575.681 644.581C585.717 639.609 590.991 628.403 588.392 617.526L556.335 482.08C552.812 467.309 547.677 452.89 540.885 439.18L480.381 317.045C461.959 279.858 412.882 272.253 384.602 300.771ZM512.211 453.385C517.893 464.855 522.231 476.99 525.137 489.389L555.696 618.658L357.037 717.072L171.27 618.267C157.613 610.972 152.13 593.823 159.202 580.164C166.184 566.55 182.885 561.29 196.543 568.585L239.392 591.328C246.221 594.975 253.761 587.669 250.298 580.679L155.391 389.099C138.522 355.048 188.653 331.442 205.344 365.134L275.836 507.43C277.79 511.373 282.614 513.001 286.556 511.048L292.918 507.896C296.861 505.943 298.489 501.119 296.536 497.176L209.619 321.725C192.75 287.674 242.881 264.067 259.572 297.76L346.134 472.494C348.087 476.437 352.911 478.064 356.854 476.111L363.216 472.959C367.158 471.006 368.786 466.182 366.833 462.24L295.675 318.599C278.806 284.548 328.937 260.942 345.628 294.634L416.52 437.737C418.473 441.679 423.297 443.307 427.24 441.354L433.602 438.202C437.545 436.249 439.172 431.425 437.219 427.482L402.15 356.693C385.459 323 434.569 297.333 451.482 331.473L511.942 453.518L512.211 453.385Z"
          fill="gray"
        />
        <path
          d="M600.843 300.62L610.964 280.189C629.963 241.837 681.632 233.954 709.791 266.095C751.593 226.317 817.553 261.225 810.244 316.751C851.667 319.639 877.861 364.532 858.729 403.153L794.051 533.711C818.323 530.446 842.875 542.386 854.677 565.418C869.812 594.789 858.362 630.967 829.171 646.414L639.745 747.201C632.916 750.848 624.732 750.923 617.832 747.505L473.503 677C463.467 672.028 573.316 628.993 575.914 618.116L548.226 487C551.749 472.229 475.208 378.693 482 364.983L505.064 316.894C523.486 279.707 572.563 272.102 600.843 300.62ZM505.064 388.948C499.382 400.418 580.906 616.601 578 629L505.064 657L628.408 716.921L814.175 618.116C827.833 610.821 833.315 593.672 826.243 580.013C819.261 566.399 802.56 561.139 788.902 568.434L746.053 591.177C739.225 594.824 731.684 587.518 735.147 580.528L830.054 388.948C846.923 354.897 796.792 331.291 780.101 364.983L709.609 507.279C707.655 511.222 702.831 512.85 698.889 510.897L692.527 507.745C688.584 505.792 686.956 500.968 688.909 497.025L775.826 321.574C792.695 287.523 742.564 263.916 725.873 297.609L639.311 472.343C637.358 476.286 632.534 477.913 628.591 475.96L622.229 472.808C618.287 470.855 616.659 466.031 618.612 462.089L689.77 318.448C706.639 284.397 656.508 260.791 639.817 294.483L568.925 437.586C566.972 441.528 562.148 443.156 558.205 441.203L551.843 438.051C547.901 436.098 546.273 431.274 548.226 427.331L583.295 356.542C599.986 322.849 550.876 297.182 533.963 331.322L505.064 388.948Z"
          fill="gray"
        />
        <rect
          fill="currentColor"
          height="28.7675"
          rx="14.3837"
          transform="rotate(54.7453 269.491 33)"
          width="150"
          x="269.491"
          y="33"
        />
        <rect
          fill="currentColor"
          height="28.7675"
          rx="14.3837"
          transform="matrix(-0.577211 0.816595 0.816595 0.577211 763 33)"
          width="150"
        />
        <rect
          fill="currentColor"
          height="28.7675"
          rx="14.3837"
          transform="matrix(-4.37114e-08 1 1 4.37114e-08 502 0)"
          width="150"
        />
        <rect
          fill="currentColor"
          height="28.7675"
          rx="14.3837"
          transform="matrix(0.577211 -0.816595 -0.816595 -0.577211 269.491 963.924)"
          width="150"
        />
        <rect
          fill="currentColor"
          height="28.7675"
          rx="14.3837"
          transform="rotate(-125.255 763 963.924)"
          width="150"
          x="763"
          y="963.924"
        />
        <rect
          fill="currentColor"
          height="28.7675"
          rx="14.3837"
          transform="rotate(-90 502 996.924)"
          width="150"
          x="502"
          y="996.924"
        />
      </svg>
    </div>
    <div
      className="Pinpoint Clap counter active"
    >
      22
    </div>
    <div
      className="Pinpoint Clap notice "
    >
      +
      0
    </div>
  </div>
</div>
`;

exports[`Test default 1`] = `
<div
  className="Pinpoint Sidebar "
>
  <div
    className="Pinpoint Author "
  >
    <span
      aria-label="Keegan avatar"
      className="Pinpoint Internal Link  "
    >
      <img
        alt=""
        className="Pinpoint Image avatar blurhash"
        decoding="async"
        height={48}
        loading="eager"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw=="
        width={48}
      />
      <span
        className="name"
      >
        Keegan
      </span>
    </span>
  </div>
  <div
    className="Pinpoint Tag Bar "
  >
    <div
      className="Pinpoint Tag Item   "
      style={
        Object {
          "backgroundColor": "var(--tag-feature-bgcolor,#B6F5B6)",
          "border": "1px solid var(--tag-bcolor, #3f7a4333)",
          "color": "var(--tag-fgcolor, #27632e)",
        }
      }
    >
      feature
    </div>
    <div
      className="Pinpoint Tag Item   "
      style={
        Object {
          "backgroundColor": "var(--tag-improvement-bgcolor,#a2eeef)",
          "border": "1px solid var(--tag-bcolor, #23747633)",
          "color": "var(--tag-fgcolor, #005e5f)",
        }
      }
    >
      improvement
    </div>
    <div
      className="Pinpoint Tag Item   "
      style={
        Object {
          "backgroundColor": "#f473da",
          "border": "1px solid var(--tag-bcolor, #72006233)",
          "color": "var(--tag-fgcolor, #ffffff)",
        }
      }
    >
      mobile
    </div>
  </div>
  <div
    className="Pinpoint Clap wrapper "
    onClick={[Function]}
  >
    <div
      className="Pinpoint Clap icon "
    >
      <svg
        fill="none"
        height="45"
        viewBox="0 0 985 997"
        width="45"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M384.602 300.771L374.481 280.34C355.482 241.988 303.813 234.105 275.654 266.246C233.852 226.468 167.893 261.376 175.201 316.902C133.778 319.79 107.584 364.684 126.716 403.304L191.394 533.862C167.122 530.597 142.57 542.537 130.768 565.569C115.633 594.94 127.083 631.118 156.274 646.565L345.7 747.352C352.529 750.999 360.713 751.074 367.613 747.656L575.681 644.581C585.717 639.609 590.991 628.403 588.392 617.526L556.335 482.08C552.812 467.309 547.677 452.89 540.885 439.18L480.381 317.045C461.959 279.858 412.882 272.253 384.602 300.771ZM512.211 453.385C517.893 464.855 522.231 476.99 525.137 489.389L555.696 618.658L357.037 717.072L171.27 618.267C157.613 610.972 152.13 593.823 159.202 580.164C166.184 566.55 182.885 561.29 196.543 568.585L239.392 591.328C246.221 594.975 253.761 587.669 250.298 580.679L155.391 389.099C138.522 355.048 188.653 331.442 205.344 365.134L275.836 507.43C277.79 511.373 282.614 513.001 286.556 511.048L292.918 507.896C296.861 505.943 298.489 501.119 296.536 497.176L209.619 321.725C192.75 287.674 242.881 264.067 259.572 297.76L346.134 472.494C348.087 476.437 352.911 478.064 356.854 476.111L363.216 472.959C367.158 471.006 368.786 466.182 366.833 462.24L295.675 318.599C278.806 284.548 328.937 260.942 345.628 294.634L416.52 437.737C418.473 441.679 423.297 443.307 427.24 441.354L433.602 438.202C437.545 436.249 439.172 431.425 437.219 427.482L402.15 356.693C385.459 323 434.569 297.333 451.482 331.473L511.942 453.518L512.211 453.385Z"
          fill="gray"
        />
        <path
          d="M600.843 300.62L610.964 280.189C629.963 241.837 681.632 233.954 709.791 266.095C751.593 226.317 817.553 261.225 810.244 316.751C851.667 319.639 877.861 364.532 858.729 403.153L794.051 533.711C818.323 530.446 842.875 542.386 854.677 565.418C869.812 594.789 858.362 630.967 829.171 646.414L639.745 747.201C632.916 750.848 624.732 750.923 617.832 747.505L473.503 677C463.467 672.028 573.316 628.993 575.914 618.116L548.226 487C551.749 472.229 475.208 378.693 482 364.983L505.064 316.894C523.486 279.707 572.563 272.102 600.843 300.62ZM505.064 388.948C499.382 400.418 580.906 616.601 578 629L505.064 657L628.408 716.921L814.175 618.116C827.833 610.821 833.315 593.672 826.243 580.013C819.261 566.399 802.56 561.139 788.902 568.434L746.053 591.177C739.225 594.824 731.684 587.518 735.147 580.528L830.054 388.948C846.923 354.897 796.792 331.291 780.101 364.983L709.609 507.279C707.655 511.222 702.831 512.85 698.889 510.897L692.527 507.745C688.584 505.792 686.956 500.968 688.909 497.025L775.826 321.574C792.695 287.523 742.564 263.916 725.873 297.609L639.311 472.343C637.358 476.286 632.534 477.913 628.591 475.96L622.229 472.808C618.287 470.855 616.659 466.031 618.612 462.089L689.77 318.448C706.639 284.397 656.508 260.791 639.817 294.483L568.925 437.586C566.972 441.528 562.148 443.156 558.205 441.203L551.843 438.051C547.901 436.098 546.273 431.274 548.226 427.331L583.295 356.542C599.986 322.849 550.876 297.182 533.963 331.322L505.064 388.948Z"
          fill="gray"
        />
        <rect
          fill="currentColor"
          height="28.7675"
          rx="14.3837"
          transform="rotate(54.7453 269.491 33)"
          width="150"
          x="269.491"
          y="33"
        />
        <rect
          fill="currentColor"
          height="28.7675"
          rx="14.3837"
          transform="matrix(-0.577211 0.816595 0.816595 0.577211 763 33)"
          width="150"
        />
        <rect
          fill="currentColor"
          height="28.7675"
          rx="14.3837"
          transform="matrix(-4.37114e-08 1 1 4.37114e-08 502 0)"
          width="150"
        />
        <rect
          fill="currentColor"
          height="28.7675"
          rx="14.3837"
          transform="matrix(0.577211 -0.816595 -0.816595 -0.577211 269.491 963.924)"
          width="150"
        />
        <rect
          fill="currentColor"
          height="28.7675"
          rx="14.3837"
          transform="rotate(-125.255 763 963.924)"
          width="150"
          x="763"
          y="963.924"
        />
        <rect
          fill="currentColor"
          height="28.7675"
          rx="14.3837"
          transform="rotate(-90 502 996.924)"
          width="150"
          x="502"
          y="996.924"
        />
      </svg>
    </div>
    <div
      className="Pinpoint Clap counter active"
    >
      22
    </div>
    <div
      className="Pinpoint Clap notice "
    >
      +
      0
    </div>
  </div>
  <div
    className="Pinpoint SocialMedia Bar "
  >
    <a
      aria-label="Facebook"
      className="Pinpoint Internal Link Pinpoint SocialMedia ShareItem FacebookShare "
      href="https://facebook.com/sharer/sharer.php?u=https%3A%2F%2Fwww.facebook.com%2FPinpoint.Engineering"
      rel="noopener noreferrer"
      target="_blank"
      title="Facebook"
    >
      <svg
        aria-hidden="true"
        data-icon="facebook-f"
        data-prefix="fab"
        focusable="false"
        height={16}
        role="img"
        viewBox="0 0 320 512"
        width={10}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
          fill="currentColor"
        />
      </svg>
    </a>
    <a
      aria-label="Twitter"
      className="Pinpoint Internal Link Pinpoint SocialMedia ShareItem TwitterShare "
      href="https://twitter.com/intent/tweet/?text=&url=https%3A%2F%2Ftwitter.com%2Fpinpoint_sw"
      rel="noopener noreferrer"
      target="_blank"
      title="Twitter"
    >
      <svg
        aria-hidden="true"
        data-icon="twitter"
        data-prefix="fab"
        focusable="false"
        height={14}
        role="img"
        viewBox="0 0 512 512"
        width={14}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
          fill="currentColor"
        />
      </svg>
    </a>
    <a
      aria-label="LinkedIn"
      className="Pinpoint Internal Link Pinpoint SocialMedia ShareItem LinkedInShare "
      href="https://www.linkedin.com/shareArticle?url=https%3A%2F%2Flinkedin.com%2Fcompany%2Fpinpoint-software&mini=true"
      rel="noopener noreferrer"
      target="_blank"
      title="LinkedIn"
    >
      <svg
        aria-hidden="true"
        data-icon="linkedin-in"
        data-prefix="fab"
        focusable="false"
        height={16}
        role="img"
        viewBox="0 0 448 512"
        width={14}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M100.28 448H7.4V148.9h92.88zM53.79 108.1C24.09 108.1 0 83.5 0 53.8a53.79 53.79 0 0 1 107.58 0c0 29.7-24.1 54.3-53.79 54.3zM447.9 448h-92.68V302.4c0-34.7-.7-79.2-48.29-79.2-48.29 0-55.69 37.7-55.69 76.7V448h-92.78V148.9h89.08v40.8h1.3c12.4-23.5 42.69-48.3 87.88-48.3 94 0 111.28 61.9 111.28 142.3V448z"
          fill="currentColor"
        />
      </svg>
    </a>
    <a
      aria-label="Email"
      className="Pinpoint Internal Link Pinpoint SocialMedia ShareItem EmailShare "
      href="mailto:hello@pinpoint.com?subject=&body="
      title="Email"
    >
      <svg
        aria-hidden="true"
        data-icon="envelope"
        data-prefix="fas"
        focusable="false"
        height={16}
        role="img"
        viewBox="0 0 512 512"
        width={16}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
          fill="currentColor"
        />
      </svg>
    </a>
  </div>
</div>
`;

exports[`Test no author 1`] = `
<div
  className="Pinpoint Sidebar "
>
  <div
    className="Pinpoint Tag Bar "
  >
    <div
      className="Pinpoint Tag Item   "
      style={
        Object {
          "backgroundColor": "var(--tag-feature-bgcolor,#B6F5B6)",
          "border": "1px solid var(--tag-bcolor, #3f7a4333)",
          "color": "var(--tag-fgcolor, #27632e)",
        }
      }
    >
      feature
    </div>
    <div
      className="Pinpoint Tag Item   "
      style={
        Object {
          "backgroundColor": "var(--tag-improvement-bgcolor,#a2eeef)",
          "border": "1px solid var(--tag-bcolor, #23747633)",
          "color": "var(--tag-fgcolor, #005e5f)",
        }
      }
    >
      improvement
    </div>
    <div
      className="Pinpoint Tag Item   "
      style={
        Object {
          "backgroundColor": "#f473da",
          "border": "1px solid var(--tag-bcolor, #72006233)",
          "color": "var(--tag-fgcolor, #ffffff)",
        }
      }
    >
      mobile
    </div>
  </div>
  <div
    className="Pinpoint Clap wrapper "
    onClick={[Function]}
  >
    <div
      className="Pinpoint Clap icon "
    >
      <svg
        fill="none"
        height="45"
        viewBox="0 0 985 997"
        width="45"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M384.602 300.771L374.481 280.34C355.482 241.988 303.813 234.105 275.654 266.246C233.852 226.468 167.893 261.376 175.201 316.902C133.778 319.79 107.584 364.684 126.716 403.304L191.394 533.862C167.122 530.597 142.57 542.537 130.768 565.569C115.633 594.94 127.083 631.118 156.274 646.565L345.7 747.352C352.529 750.999 360.713 751.074 367.613 747.656L575.681 644.581C585.717 639.609 590.991 628.403 588.392 617.526L556.335 482.08C552.812 467.309 547.677 452.89 540.885 439.18L480.381 317.045C461.959 279.858 412.882 272.253 384.602 300.771ZM512.211 453.385C517.893 464.855 522.231 476.99 525.137 489.389L555.696 618.658L357.037 717.072L171.27 618.267C157.613 610.972 152.13 593.823 159.202 580.164C166.184 566.55 182.885 561.29 196.543 568.585L239.392 591.328C246.221 594.975 253.761 587.669 250.298 580.679L155.391 389.099C138.522 355.048 188.653 331.442 205.344 365.134L275.836 507.43C277.79 511.373 282.614 513.001 286.556 511.048L292.918 507.896C296.861 505.943 298.489 501.119 296.536 497.176L209.619 321.725C192.75 287.674 242.881 264.067 259.572 297.76L346.134 472.494C348.087 476.437 352.911 478.064 356.854 476.111L363.216 472.959C367.158 471.006 368.786 466.182 366.833 462.24L295.675 318.599C278.806 284.548 328.937 260.942 345.628 294.634L416.52 437.737C418.473 441.679 423.297 443.307 427.24 441.354L433.602 438.202C437.545 436.249 439.172 431.425 437.219 427.482L402.15 356.693C385.459 323 434.569 297.333 451.482 331.473L511.942 453.518L512.211 453.385Z"
          fill="gray"
        />
        <path
          d="M600.843 300.62L610.964 280.189C629.963 241.837 681.632 233.954 709.791 266.095C751.593 226.317 817.553 261.225 810.244 316.751C851.667 319.639 877.861 364.532 858.729 403.153L794.051 533.711C818.323 530.446 842.875 542.386 854.677 565.418C869.812 594.789 858.362 630.967 829.171 646.414L639.745 747.201C632.916 750.848 624.732 750.923 617.832 747.505L473.503 677C463.467 672.028 573.316 628.993 575.914 618.116L548.226 487C551.749 472.229 475.208 378.693 482 364.983L505.064 316.894C523.486 279.707 572.563 272.102 600.843 300.62ZM505.064 388.948C499.382 400.418 580.906 616.601 578 629L505.064 657L628.408 716.921L814.175 618.116C827.833 610.821 833.315 593.672 826.243 580.013C819.261 566.399 802.56 561.139 788.902 568.434L746.053 591.177C739.225 594.824 731.684 587.518 735.147 580.528L830.054 388.948C846.923 354.897 796.792 331.291 780.101 364.983L709.609 507.279C707.655 511.222 702.831 512.85 698.889 510.897L692.527 507.745C688.584 505.792 686.956 500.968 688.909 497.025L775.826 321.574C792.695 287.523 742.564 263.916 725.873 297.609L639.311 472.343C637.358 476.286 632.534 477.913 628.591 475.96L622.229 472.808C618.287 470.855 616.659 466.031 618.612 462.089L689.77 318.448C706.639 284.397 656.508 260.791 639.817 294.483L568.925 437.586C566.972 441.528 562.148 443.156 558.205 441.203L551.843 438.051C547.901 436.098 546.273 431.274 548.226 427.331L583.295 356.542C599.986 322.849 550.876 297.182 533.963 331.322L505.064 388.948Z"
          fill="gray"
        />
        <rect
          fill="currentColor"
          height="28.7675"
          rx="14.3837"
          transform="rotate(54.7453 269.491 33)"
          width="150"
          x="269.491"
          y="33"
        />
        <rect
          fill="currentColor"
          height="28.7675"
          rx="14.3837"
          transform="matrix(-0.577211 0.816595 0.816595 0.577211 763 33)"
          width="150"
        />
        <rect
          fill="currentColor"
          height="28.7675"
          rx="14.3837"
          transform="matrix(-4.37114e-08 1 1 4.37114e-08 502 0)"
          width="150"
        />
        <rect
          fill="currentColor"
          height="28.7675"
          rx="14.3837"
          transform="matrix(0.577211 -0.816595 -0.816595 -0.577211 269.491 963.924)"
          width="150"
        />
        <rect
          fill="currentColor"
          height="28.7675"
          rx="14.3837"
          transform="rotate(-125.255 763 963.924)"
          width="150"
          x="763"
          y="963.924"
        />
        <rect
          fill="currentColor"
          height="28.7675"
          rx="14.3837"
          transform="rotate(-90 502 996.924)"
          width="150"
          x="502"
          y="996.924"
        />
      </svg>
    </div>
    <div
      className="Pinpoint Clap counter active"
    >
      22
    </div>
    <div
      className="Pinpoint Clap notice "
    >
      +
      0
    </div>
  </div>
</div>
`;

exports[`clap only 1`] = `
<div
  className="Pinpoint Sidebar "
>
  <div
    className="Pinpoint Tag Bar "
  >
    <div
      className="Pinpoint Tag Item   "
      style={
        Object {
          "backgroundColor": "var(--tag-feature-bgcolor,#B6F5B6)",
          "border": "1px solid var(--tag-bcolor, #3f7a4333)",
          "color": "var(--tag-fgcolor, #27632e)",
        }
      }
    >
      feature
    </div>
    <div
      className="Pinpoint Tag Item   "
      style={
        Object {
          "backgroundColor": "var(--tag-improvement-bgcolor,#a2eeef)",
          "border": "1px solid var(--tag-bcolor, #23747633)",
          "color": "var(--tag-fgcolor, #005e5f)",
        }
      }
    >
      improvement
    </div>
    <div
      className="Pinpoint Tag Item   "
      style={
        Object {
          "backgroundColor": "#f473da",
          "border": "1px solid var(--tag-bcolor, #72006233)",
          "color": "var(--tag-fgcolor, #ffffff)",
        }
      }
    >
      mobile
    </div>
  </div>
  <div
    className="Pinpoint Clap wrapper "
    onClick={[Function]}
  >
    <div
      className="Pinpoint Clap icon "
    >
      <svg
        fill="none"
        height="45"
        viewBox="0 0 985 997"
        width="45"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M384.602 300.771L374.481 280.34C355.482 241.988 303.813 234.105 275.654 266.246C233.852 226.468 167.893 261.376 175.201 316.902C133.778 319.79 107.584 364.684 126.716 403.304L191.394 533.862C167.122 530.597 142.57 542.537 130.768 565.569C115.633 594.94 127.083 631.118 156.274 646.565L345.7 747.352C352.529 750.999 360.713 751.074 367.613 747.656L575.681 644.581C585.717 639.609 590.991 628.403 588.392 617.526L556.335 482.08C552.812 467.309 547.677 452.89 540.885 439.18L480.381 317.045C461.959 279.858 412.882 272.253 384.602 300.771ZM512.211 453.385C517.893 464.855 522.231 476.99 525.137 489.389L555.696 618.658L357.037 717.072L171.27 618.267C157.613 610.972 152.13 593.823 159.202 580.164C166.184 566.55 182.885 561.29 196.543 568.585L239.392 591.328C246.221 594.975 253.761 587.669 250.298 580.679L155.391 389.099C138.522 355.048 188.653 331.442 205.344 365.134L275.836 507.43C277.79 511.373 282.614 513.001 286.556 511.048L292.918 507.896C296.861 505.943 298.489 501.119 296.536 497.176L209.619 321.725C192.75 287.674 242.881 264.067 259.572 297.76L346.134 472.494C348.087 476.437 352.911 478.064 356.854 476.111L363.216 472.959C367.158 471.006 368.786 466.182 366.833 462.24L295.675 318.599C278.806 284.548 328.937 260.942 345.628 294.634L416.52 437.737C418.473 441.679 423.297 443.307 427.24 441.354L433.602 438.202C437.545 436.249 439.172 431.425 437.219 427.482L402.15 356.693C385.459 323 434.569 297.333 451.482 331.473L511.942 453.518L512.211 453.385Z"
          fill="gray"
        />
        <path
          d="M600.843 300.62L610.964 280.189C629.963 241.837 681.632 233.954 709.791 266.095C751.593 226.317 817.553 261.225 810.244 316.751C851.667 319.639 877.861 364.532 858.729 403.153L794.051 533.711C818.323 530.446 842.875 542.386 854.677 565.418C869.812 594.789 858.362 630.967 829.171 646.414L639.745 747.201C632.916 750.848 624.732 750.923 617.832 747.505L473.503 677C463.467 672.028 573.316 628.993 575.914 618.116L548.226 487C551.749 472.229 475.208 378.693 482 364.983L505.064 316.894C523.486 279.707 572.563 272.102 600.843 300.62ZM505.064 388.948C499.382 400.418 580.906 616.601 578 629L505.064 657L628.408 716.921L814.175 618.116C827.833 610.821 833.315 593.672 826.243 580.013C819.261 566.399 802.56 561.139 788.902 568.434L746.053 591.177C739.225 594.824 731.684 587.518 735.147 580.528L830.054 388.948C846.923 354.897 796.792 331.291 780.101 364.983L709.609 507.279C707.655 511.222 702.831 512.85 698.889 510.897L692.527 507.745C688.584 505.792 686.956 500.968 688.909 497.025L775.826 321.574C792.695 287.523 742.564 263.916 725.873 297.609L639.311 472.343C637.358 476.286 632.534 477.913 628.591 475.96L622.229 472.808C618.287 470.855 616.659 466.031 618.612 462.089L689.77 318.448C706.639 284.397 656.508 260.791 639.817 294.483L568.925 437.586C566.972 441.528 562.148 443.156 558.205 441.203L551.843 438.051C547.901 436.098 546.273 431.274 548.226 427.331L583.295 356.542C599.986 322.849 550.876 297.182 533.963 331.322L505.064 388.948Z"
          fill="gray"
        />
        <rect
          fill="currentColor"
          height="28.7675"
          rx="14.3837"
          transform="rotate(54.7453 269.491 33)"
          width="150"
          x="269.491"
          y="33"
        />
        <rect
          fill="currentColor"
          height="28.7675"
          rx="14.3837"
          transform="matrix(-0.577211 0.816595 0.816595 0.577211 763 33)"
          width="150"
        />
        <rect
          fill="currentColor"
          height="28.7675"
          rx="14.3837"
          transform="matrix(-4.37114e-08 1 1 4.37114e-08 502 0)"
          width="150"
        />
        <rect
          fill="currentColor"
          height="28.7675"
          rx="14.3837"
          transform="matrix(0.577211 -0.816595 -0.816595 -0.577211 269.491 963.924)"
          width="150"
        />
        <rect
          fill="currentColor"
          height="28.7675"
          rx="14.3837"
          transform="rotate(-125.255 763 963.924)"
          width="150"
          x="763"
          y="963.924"
        />
        <rect
          fill="currentColor"
          height="28.7675"
          rx="14.3837"
          transform="rotate(-90 502 996.924)"
          width="150"
          x="502"
          y="996.924"
        />
      </svg>
    </div>
    <div
      className="Pinpoint Clap counter active"
    >
      22
    </div>
    <div
      className="Pinpoint Clap notice "
    >
      +
      0
    </div>
  </div>
</div>
`;
